phase2/kubeadm/do

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+set -x
+
+cd "${BASH_SOURCE%/*}"
+
+CONFIG_FILE="${CONFIG_JSON_FILE:-../../.config.json}"
+KUBEADM_INIT_PARAM_FILE=/etc/kubeadm/kubeadm_init_params.txt
+
+upgrade_master() {
+  KUBERNETES_VERSION=$(jq -r '.phase2.kubernetes_version' ${CONFIG_FILE})
+  UPGRADE_METHOD=$(jq -r '.phase2.kubeadm.master_upgrade.method' ${CONFIG_FILE})
```

Comment:
Per @luxas' suggestion, instead of having `.phase2.kubeadm.*` for upgrade options, they do seem generic enough to move to just `.phase2` (e.g. `.phase2.upgrade_strategy`).

Also, is there a need for having the upgrade method scoped to the master here? I'm having trouble imagining a scenario where a normal user would want to use different upgrade strategies on the master versus nodes, so I'm wondering if there are specific test scenarios you're trying to enable.

Reply:
This config is specifically how do you want kubeadm to upgrade the master/control plane. It has no relevance to how the node is updated. It is also not an overall upgrade strategy.

I can see your point if it was an name for an overall upgrade strategy or if it was a setting that would make sense for other phase2 providers (eg. should I drain the node before update).